### ISSUE TITLE
Implement Index<Range>, Index<RangeFrom>, Index<RangeTo>, Index<RangeFull>

### ISSUE THREAD

**Issue by Geal** [contain-rs/bit-vec#13]
Hi!

People are asking for bit level parsing in [nom](https://github.com/Geal/nom), and to use `BitVec`, I would need to implement those traits to get the slicing syntax on bit streams.

I see in the code that `Index<usize>` is already implemented, but there's [a big warning](https://github.com/contain-rs/bit-vec/blob/master/src/lib.rs#L199). Could you explain what it means?

I can try to implement those, but I wonder if you would accept it (because of that warning)

**Comment by Gankro** [comment]
Ah, hehe. The NOPE NOPE NOPE is me being upset that we need to "hack" around the Index trait's design by outputting references to TRUE or FALSE *statics*. It's workable; just stupid.

**Comment by Gankro** [comment]
That said, I don't think you can use indexing for bit-ranges. You can't return an `&[u1]`, you'll have to return an iterator.

**Comment by Geal** [comment]
yes, it makes sense. Since nom uses a lot of macros, I just need to trick it into thinking it is a real slice. Maybe by returning a `Vec<bool>` generated from the specified range?

**Comment by Geal** [comment]
I'm not sure adding it to this crate would be worth the trouble if it is only used in nom. But then, my only option is to duplicate the code, since I can't implement the traits on `BitVec` from another crate :(

**Comment by Gankro** [comment]
How would you return a Vec<bool>? Indexing requires that you return a reference.
